32-bit x86 Runtime Revamp

MingcongBai opened the pull request:
Topic Description
-----------------

- x264+32: fix a typo H264 => H.264
- nasm+32: drop, orphaned
- x11-lib+32: drop invalid PKGREP/PKGPROV
- wavpack+32: drop unneeded AUTOTOOLS_AFTER
- vulkan-tools+32: use array for CMAKE_AFTER
- vulkan-headers+32: drop trailing space in PKGDEP
- tcl+32: fix a typo Lanuage => Language
- sqlite+32: drop advertisement-like description
- sqlite+32: add comments for fixes and hacks
- spirv-tools+32: use array for CMAKE_AFTER

... and 770 more commits

Package(s) Affected
-------------------

- 32subsystem: 1:3
- a52dec+32: 0.7.4-3
- aalib+32: 1:1.4~rc5
- acl+32: 2.3.1
- alsa-lib+32: 1.2.13
- alsa-plugins+32: 1.2.12
- aosc-aaa+32: 12.0.4
- at-spi2-core+32: 2.52.0
- attr+32: 2.4.48-3
- autobuild4: 4.7.9+git20250219
- binutils+32: 2.43.1
- brotli+32: 1.1.0
- bzip2+32: 1.0.8-2
- cairo+32: 1.18.2-1
- cdparanoia+32: 10.2-4
- celt+32: 0.11.3-3
- cups+32: 2.4.11
- curl+32: 8.12.1
- dbus+32: 1.14.10
- dbus-glib+32: 0.110
- devel-base+32: 0-1
- elfutils+32: 0.191
- enca+32: 1.19-3
- eudev+32: 3.2.14
- expat+32: 2.6.2-1
- faac+32: 1.29.9.2
- faad2+32: 2.11.1
- ffmpeg+32: 7.1
- fftw+32: 3.3.10
- flac+32: 1.4.3-1
- fluidsynth+32: 2.3.6
- fontconfig+32: 2.15.0
- freealut+32: 1.1.0-3
- freeglut+32: 3.4.0
- freetype+32: 2.13.3
- fribidi+32: 1.0.14
- game-music-emu+32: 0.6.3
- gcc+32: 14.2.0
- gconf+32: 3.2.6-3
- gdbm+32: 1.24
- gdk-pixbuf+32: 2.42.12
- giflib+32: 5.1.8-1
- glew+32: 2.2.0-2
- glib+32: 2.83.3
- glibc+32: 2.40
- glslang+32: 13.1.1
- glu+32: 9.0.3
- gmp+32: 6.3.0-1
- gnutls+32: 3.8.7
- gsm+32: 1.0.19-1
- gtk-2+32: 2.24.33
- gtk-3+32: 3.24.43
- harfbuzz+32: 8.4.0
- icu+32: 74.2
- imlib2+32: 1.5.1
- json-c+32: 0.18
- krb5+32: 1.21.3
- lame+32: 3.100-2
- lcms2+32: 2.14
- libaio+32: 0.3.113
- libass+32: 0.17.2
- libasyncns+32: 0.8-5
- libavc1394+32: 0.5.4-4
- libbluray+32: 1.3.4
- libbsd+32: 0.12.2
- libcaca+32: 1:0.99.beta19-5
- libcap+32: 2.73
- libcddb+32: 1.3.2-3
- libcdio+32: 2.1.0
- libcl+32: 2.3.2
- libclc+32: 18.1.8
- libcroco+32: 0.6.13-1
- libdatrie+32: 0.2.13
- libdca+32: 0.0.7-1
- libdrm+32: 2.4.124
- libdv+32: 1.0.0-3
- libdvdcss+32: 1.4.3
- libdvdnav+32: 6.1.1
- libdvdread+32: 6.1.3
- libepoxy+32: 1.5.10
- libexif+32: 0.6.22-1
- libffi+32: 3.4.7
- libgcrypt+32: 1.11.0
- libglvnd+32: 1.7.0
- libgpg-error+32: 1.50
- libid3tag+32: 0.15.1b-6
- libidn+32: 1.42
- libiec61883+32: 1.2.0-3
- libjpeg-turbo+32: 3.0.4
- libmad+32: 0.15.1b-6
- libmd+32: 1.1.0
- libmikmod+32: 3.3.11.1-1
- libmms+32: 0.6.4-3
- libmodplug+32: 0.8.8.5-3
- libmpeg2+32: 0.5.1-3
- libnl+32: 3.9.0
- libofa+32: 0.9.3-3
- libogg+32: 1.3.4-2
- libpcap+32: 1.10.4
- libpng+32: 2:1.6.44
- libpng-1.6+32: 2:0
- libraw1394+32: 2.1.2-3
- librsvg+32: 2.59.2
- libshout+32: 2.4.3-1
- libsndfile+32: 1.2.0-2
- libssh+32: 0.8.9-3
- libssh2+32: 1.9.0-3
- libtasn1+32: 4.20.0
- libthai+32: 0.1.28
- libtheora+32: 1.1.1-4
- libtiff+32: 4.4.0
- libtool+32: 2.5.4
- libunwind+32: 1.8.1
- libusb+32: 1.0.23-2
- libusb-compat+32: 0.1.5-2
- libva+32: 2.22.0
- libva-intel-driver+32: 2.4.1-1
- libva-vdpau-driver+32: 0.7.4-4
- libvdpau+32: 1.5
- libvorbis+32: 1.3.7-1
- libvpx+32: 1.15.0-1
- libwebp+32: 1.3.1
- libxau+32: 1.0.12
- libxcb+32: 1.17.0
- libxcrypt+32: 4.4.37
- libxdmcp+32: 1.1.5
- libxkbcommon+32: 1.7.0
- libxml2+32: 2.10.4-1
- libxnvctrl: 565.57.01-1
- libxnvctrl+32: 565.57.01
- libxslt+32: 1.1.42
- linux+api+32: 6.12.7
- llvm+32: 18.1.8
- lm-sensors+32: 3.6.0
- lzo+32: 2.10
- mangohud+32: 0.8.0
- mbedtls+32: 2.16.7-1
- mesa+32: 24.3.4+dxheaders1.614.1
- mjpegtools+32: 2.1.0-3
- mp4v2+32: 4.1.3
- mpc+32: 1.3.1-1
- mpfr+32: 4.2.1-1
- mpg123+32: 1.32.8-1
- ncurses+32: 6.4-1
- neon+32: 0.33.0
- nettle+32: 3.10.1
- nlohmann-json+32: 3.11.3-1
- nspr+32: 4.35-1
- nss+32: 3.107
- nvidia-open: 565.77+utils565.57.01-1
- openal-soft+32: 1.23.1-1
- opencl-registry-api+32: 2024.05.08
- opencore-amr+32: 0.1.5-3
- openjpeg-legacy+32: 1.5.2-3
- openldap+32: 2.6.9
- openssl+32: 3.4.1-1
- openssl-1.1+32: 1.1.1w
- opus+32: 1.3.1-2
- orc+32: 0.4.32-1
- p11-kit+32: 0.25.5
- pango+32: 1.56.1
- pcre+32: 8.45
- pcre2+32: 10.44
- pixman+32: 0.43.0
- popt+32: 1.18
- portaudio+32: 19.7.0
- proxychains-ng+32: 1:4.17
- pulseaudio+32: 17.0
- readline+32: 8.2.1-1
- recode+32: 3.7.14
- rtmpdump+32: 20151223
- rustc+32: 1.84.1
- s2tc+32: 1:1.0+git20210317
- sbc+32: 2.0
- schroedinger+32: 1.0.11-3
- sdl+32: 1.2.15-9
- sdl-ttf+32: 2.0.11-3
- sdl2+32: 2.30.4
- shared-mime-info+32: 2.4
- smpeg+32: 0.4.5-3
- soundtouch+32: 2.3.3
- soxr+32: 0.1.3-5
- speex+32: 1:1.2.0-3
- spirv-llvm-translator+32: 18.1.0
- spirv-tools+32: 1:1.3.283.0
- sqlite+32: 3.46.1
- taglib+32: 1.13.1
- tcl+32: 8.6.14
- tslib+32: 1.21
- twolame+32: 0.3.13-4
- util-linux+32: 2.40.4
- v4l-utils+32: 1.22.1
- vkd3d+32: 1.9-1
- volk-meta-loader+32: 1.3.283.0-1
- vulkan+32: 1:0-2
- vulkan-headers+32: 1.3.283.0-1
- vulkan-loader+32: 1.3.283.0-1
- vulkan-tools+32: 1.3.283.0-1
- wavpack+32: 5.8.1
- wayland+32: 1.23.1
- wildmidi+32: 0.4.6
- wine: 3:10.1+gecko2.47.4+mono9.4.0
- x11-lib+32: 7.7.20240731
- x11-proto+32: 2024.1
- x264+32: 1:0+git20241027
- x265+32: 3.6
- xvidcore+32: 1.3.7-3
- xz+32: 5.6.3
- zlib+32: 2.2.3
- zstd+32: 1.5.6

Security Update?
----------------

No

Build Order
-----------

```
#buildit autobuild4 nvidia-open libxnvctrl aosc-aaa+32:-pkgbreak groups/optenv32 wine aosc-aaa+32
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [ ] AArch64 `arm64`
- [ ] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

MingcongBai committed:
fontconfig+32: depend on aosc-aaa+32

MingcongBai committed:
game-music-emu+32: depend on aosc-aaa+32

MingcongBai committed:
giflib+32: depend on aosc-aaa+32

MingcongBai committed:
freetype+32: depend on aosc-aaa+32

MingcongBai committed:
glslang+32: depend on aosc-aaa+32

MingcongBai committed:
freeglut+32: depend on aosc-aaa+32

MingcongBai committed:
glib+32: depend on aosc-aaa+32

MingcongBai committed:
freealut+32: depend on aosc-aaa+32

MingcongBai committed:
gcc+32: depend on aosc-aaa+32

MingcongBai committed:
flac+32: depend on aosc-aaa+32

MingcongBai committed:
gdbm+32: depend on aosc-aaa+32

MingcongBai committed:
glew+32: depend on aosc-aaa+32

MingcongBai committed:
fribidi+32: depend on aosc-aaa+32

MingcongBai committed:
glibc+32: depend on aosc-aaa+32

MingcongBai committed:
gconf+32: depend on aosc-aaa+32

MingcongBai committed:
fluidsynth+32: depend on aosc-aaa+32

MingcongBai committed:
gdk-pixbuf+32: depend on aosc-aaa+32

MingcongBai committed:
gsm+32: depend on aosc-aaa+32

MingcongBai committed:
libavc1394+32: depend on aosc-aaa+32

MingcongBai committed:
imlib2+32: depend on aosc-aaa+32

MingcongBai committed:
glu+32: depend on aosc-aaa+32

MingcongBai committed:
libbluray+32: depend on aosc-aaa+32

MingcongBai committed:
gtk-2+32: depend on aosc-aaa+32

MingcongBai committed:
libbsd+32: depend on aosc-aaa+32

MingcongBai committed:
libaio+32: depend on aosc-aaa+32

MingcongBai committed:
gmp+32: depend on aosc-aaa+32

MingcongBai committed:
harfbuzz+32: depend on aosc-aaa+32

MingcongBai committed:
krb5+32: depend on aosc-aaa+32

MingcongBai committed:
libasyncns+32: depend on aosc-aaa+32

MingcongBai committed:
gnutls+32: depend on aosc-aaa+32

MingcongBai committed:
libass+32: depend on aosc-aaa+32

MingcongBai committed:
gtk-3+32: depend on aosc-aaa+32

MingcongBai committed:
lcms2+32: depend on aosc-aaa+32

MingcongBai committed:
json-c+32: depend on aosc-aaa+32

MingcongBai committed:
lame+32: depend on aosc-aaa+32

MingcongBai committed:
libgcrypt+32: depend on aosc-aaa+32

MingcongBai committed:
libdvdnav+32: depend on aosc-aaa+32

MingcongBai committed:
libdvdread+32: depend on aosc-aaa+32

MingcongBai committed:
libclc+32: depend on aosc-aaa+32

MingcongBai committed:
libepoxy+32: depend on aosc-aaa+32

MingcongBai committed:
libcl+32: depend on aosc-aaa+32

MingcongBai committed:
libcdio+32: depend on aosc-aaa+32

MingcongBai committed:
libglvnd+32: depend on aosc-aaa+32

MingcongBai committed:
libdv+32: depend on aosc-aaa+32

MingcongBai committed:
libdrm+32: depend on aosc-aaa+32

MingcongBai committed:
libdatrie+32: depend on aosc-aaa+32

MingcongBai committed:
libcroco+32: depend on aosc-aaa+32

MingcongBai committed:
libexif+32: depend on aosc-aaa+32

MingcongBai committed:
libcddb+32: depend on aosc-aaa+32

MingcongBai committed:
libdca+32: depend on aosc-aaa+32

MingcongBai committed:
libdvdcss+32: depend on aosc-aaa+32

MingcongBai committed:
libcap+32: depend on aosc-aaa+32

MingcongBai committed:
libcaca+32: depend on aosc-aaa+32

MingcongBai committed:
libffi+32: depend on aosc-aaa+32

MingcongBai committed:
libid3tag+32: depend on aosc-aaa+32

MingcongBai committed:
libraw1394+32: depend on aosc-aaa+32

MingcongBai committed:
libpng-1.6+32: depend on aosc-aaa+32

MingcongBai committed:
libgpg-error+32: depend on aosc-aaa+32

MingcongBai committed:
libmd+32: depend on aosc-aaa+32

MingcongBai committed:
libiec61883+32: depend on aosc-aaa+32

MingcongBai committed:
libmodplug+32: depend on aosc-aaa+32

MingcongBai committed:
libjpeg-turbo+32: depend on aosc-aaa+32

MingcongBai committed:
libmikmod+32: depend on aosc-aaa+32

MingcongBai committed:
libmpeg2+32: depend on aosc-aaa+32

MingcongBai committed:
libidn+32: depend on aosc-aaa+32

MingcongBai committed:
libmad+32: depend on aosc-aaa+32

MingcongBai committed:
libnl+32: depend on aosc-aaa+32

MingcongBai committed:
libofa+32: depend on aosc-aaa+32

MingcongBai committed:
libmms+32: depend on aosc-aaa+32

MingcongBai committed:
libpcap+32: depend on aosc-aaa+32

MingcongBai committed:
libpng+32: depend on aosc-aaa+32

MingcongBai committed:
libogg+32: depend on aosc-aaa+32

MingcongBai committed:
libshout+32: depend on aosc-aaa+32

MingcongBai committed:
libva+32: depend on aosc-aaa+32

MingcongBai committed:
libva-vdpau-driver+32: depend on aosc-aaa+32

MingcongBai committed:
libtool+32: depend on aosc-aaa+32

MingcongBai committed:
libusb+32: depend on aosc-aaa+32

MingcongBai committed:
libssh+32: depend on aosc-aaa+32

MingcongBai committed:
librsvg+32: depend on aosc-aaa+32

MingcongBai committed:
libthai+32: depend on aosc-aaa+32

MingcongBai committed:
libvdpau+32: depend on aosc-aaa+32

MingcongBai committed:
libssh2+32: depend on aosc-aaa+32

MingcongBai committed:
libvpx+32: depend on aosc-aaa+32

MingcongBai committed:
libva-intel-driver+32: depend on aosc-aaa+32

MingcongBai committed:
libtiff+32: depend on aosc-aaa+32

MingcongBai committed:
libvorbis+32: depend on aosc-aaa+32

MingcongBai committed:
libtheora+32: depend on aosc-aaa+32

MingcongBai committed:
libtasn1+32: depend on aosc-aaa+32

MingcongBai committed:
libusb-compat+32: depend on aosc-aaa+32

MingcongBai committed:
libunwind+32: depend on aosc-aaa+32

MingcongBai committed:
libsndfile+32: depend on aosc-aaa+32

MingcongBai committed:
mesa+32: depend on aosc-aaa+32

MingcongBai committed:
mjpegtools+32: depend on aosc-aaa+32

MingcongBai committed:
libxslt+32: depend on aosc-aaa+32

MingcongBai committed:
libxnvctrl+32: depend on aosc-aaa+32

MingcongBai committed:
mp4v2+32: depend on aosc-aaa+32

MingcongBai committed:
libxau+32: depend on aosc-aaa+32

MingcongBai committed:
mangohud+32: depend on aosc-aaa+32

MingcongBai committed:
libxkbcommon+32: depend on aosc-aaa+32

MingcongBai committed:
mbedtls+32: depend on aosc-aaa+32

MingcongBai committed:
libxml2+32: depend on aosc-aaa+32

MingcongBai committed:
libxcb+32: depend on aosc-aaa+32

MingcongBai committed:
lzo+32: depend on aosc-aaa+32

MingcongBai committed:
libwebp+32: depend on aosc-aaa+32

MingcongBai committed:
mpc+32: depend on aosc-aaa+32

MingcongBai committed:
lm-sensors+32: depend on aosc-aaa+32

MingcongBai committed:
libxdmcp+32: depend on aosc-aaa+32

MingcongBai committed:
llvm+32: depend on aosc-aaa+32

MingcongBai committed:
libxcrypt+32: depend on aosc-aaa+32

MingcongBai committed:
opencl-registry-api+32: depend on aosc-aaa+32

MingcongBai committed:
opus+32: depend on aosc-aaa+32

MingcongBai committed:
openssl+32: depend on aosc-aaa+32

MingcongBai committed:
openal-soft+32: depend on aosc-aaa+32

MingcongBai committed:
nss+32: depend on aosc-aaa+32

MingcongBai committed:
openssl-1.1+32: depend on aosc-aaa+32

MingcongBai committed:
mpfr+32: depend on aosc-aaa+32

MingcongBai committed:
orc+32: depend on aosc-aaa+32

MingcongBai committed:
openjpeg-legacy+32: depend on aosc-aaa+32

MingcongBai committed:
nspr+32: depend on aosc-aaa+32

MingcongBai committed:
nlohmann-json+32: depend on aosc-aaa+32

MingcongBai committed:
nasm+32: depend on aosc-aaa+32

MingcongBai committed:
ncurses+32: depend on aosc-aaa+32

MingcongBai committed:
opencore-amr+32: depend on aosc-aaa+32

MingcongBai committed:
nettle+32: depend on aosc-aaa+32

MingcongBai committed:
mpg123+32: depend on aosc-aaa+32

MingcongBai committed:
openldap+32: depend on aosc-aaa+32

MingcongBai committed:
p11-kit+32: depend on aosc-aaa+32

MingcongBai committed:
neon+32: depend on aosc-aaa+32

MingcongBai committed:
pulseaudio+32: depend on aosc-aaa+32

MingcongBai committed:
popt+32: depend on aosc-aaa+32

MingcongBai committed:
s2tc+32: depend on aosc-aaa+32

MingcongBai committed:
pcre2+32: depend on aosc-aaa+32

MingcongBai committed:
readline+32: depend on aosc-aaa+32

MingcongBai committed:
sdl+32: depend on aosc-aaa+32

MingcongBai committed:
sdl-ttf+32: depend on aosc-aaa+32

MingcongBai committed:
rtmpdump+32: depend on aosc-aaa+32

MingcongBai committed:
rustc+32: depend on aosc-aaa+32

MingcongBai committed:
pixman+32: depend on aosc-aaa+32

MingcongBai committed:
pcre+32: depend on aosc-aaa+32

MingcongBai committed:
pango+32: depend on aosc-aaa+32

MingcongBai committed:
sbc+32: depend on aosc-aaa+32

MingcongBai committed:
recode+32: depend on aosc-aaa+32

MingcongBai committed:
schroedinger+32: depend on aosc-aaa+32

MingcongBai committed:
proxychains-ng+32: depend on aosc-aaa+32

MingcongBai committed:
sdl2+32: depend on aosc-aaa+32

MingcongBai committed:
portaudio+32: depend on aosc-aaa+32

MingcongBai committed:
smpeg+32: depend on aosc-aaa+32

MingcongBai committed:
vkd3d+32: depend on aosc-aaa+32

MingcongBai committed:
twolame+32: depend on aosc-aaa+32

MingcongBai committed:
tslib+32: depend on aosc-aaa+32

MingcongBai committed:
sqlite+32: depend on aosc-aaa+32

MingcongBai committed:
tcl+32: depend on aosc-aaa+32

MingcongBai committed:
spirv-llvm-translator+32: depend on aosc-aaa+32

MingcongBai committed:
vulkan-headers+32: depend on aosc-aaa+32

MingcongBai committed:
spirv-tools+32: depend on aosc-aaa+32

MingcongBai committed:
soxr+32: depend on aosc-aaa+32

MingcongBai committed:
vulkan+32: depend on aosc-aaa+32

MingcongBai committed:
volk-meta-loader+32: depend on aosc-aaa+32

MingcongBai committed:
taglib+32: depend on aosc-aaa+32

MingcongBai committed:
speex+32: depend on aosc-aaa+32

MingcongBai committed:
soundtouch+32: depend on aosc-aaa+32

MingcongBai committed:
util-linux+32: depend on aosc-aaa+32

MingcongBai committed:
v4l-utils+32: depend on aosc-aaa+32

MingcongBai committed:
vulkan-loader+32: depend on aosc-aaa+32

MingcongBai committed:
shared-mime-info+32: depend on aosc-aaa+32

MingcongBai committed:
wavpack+32: depend on aosc-aaa+32

MingcongBai committed:
x11-proto+32: depend on aosc-aaa+32

MingcongBai committed:
xvidcore+32: depend on aosc-aaa+32

MingcongBai committed:
x264+32: depend on aosc-aaa+32

MingcongBai committed:
vulkan-tools+32: depend on aosc-aaa+32

MingcongBai committed:
wayland+32: depend on aosc-aaa+32

MingcongBai committed:
wildmidi+32: depend on aosc-aaa+32

MingcongBai committed:
x265+32: depend on aosc-aaa+32

MingcongBai committed:
zstd+32: depend on aosc-aaa+32

MingcongBai committed:
zlib+32: depend on aosc-aaa+32

MingcongBai committed:
x11-lib+32: depend on aosc-aaa+32

MingcongBai committed:
xz+32: depend on aosc-aaa+32

MingcongBai committed:
wavpack+32: correct as optenv32 ABHOST

MingcongBai committed:
twolame+32: correct as optenv32 ABHOST

MingcongBai committed:
aosc-aaa+32: break and replace all previous +32 packages

MingcongBai committed:
libpng+32: update last libpng-1.6+32 version

MingcongBai committed:
libxslt+32: drop python-2+32 dep

MingcongBai committed:
openssl+32: bump REL against stable

MingcongBai committed:
32subsystem: drop orphaned packages

MingcongBai committed:
fontconfig+32: use shared /etc configuration

MingcongBai committed:
mpfr+32: drop hard-coded docdir option

Cyanoxygen committed:
gcc+32: inject the correct path to as and ld

GCC will launch the correct linker and assembler in /opt/32 instead of
/usr/bin, simulating the `-B' option.

MingcongBai committed:
gcc+32: drop hard-coded mandir, infodir

MingcongBai committed:
glibc+32: drop hard-coded /etc

MingcongBai committed:
lm-sensors: set ETCDIR to /opt/32/etc

Leverage the shared /opt/32/etc => /etc symlink.

MingcongBai committed:
util-linux+32: drop unneeded shared data

MingcongBai committed:
libxkbcommon+32: drop unneeded shared data

MingcongBai committed:
p11-kit+32: drop unneeded shared data

MingcongBai committed:
nss+32: skip installation of shared data

MingcongBai committed:
pulseaudio+32: drop unneeded shared data

MingcongBai committed:
gconf+32: set SETUPTOOLS_USE_DISTUTILS=1 for newer setuptools

MingcongBai committed:
cups+32: drop unneeded configuration files

MingcongBai committed:
cups+32: drop unneeded shared data

MingcongBai committed:
harfbuzz+32: set SETUPTOOLS_USE_DISTUTILS=1 for newer setuptools

OriginCode commented on runtime-optenv32/aalib+32/autobuild/defines:
```diff
@@ -1,9 +1,17 @@
+
+ABHOST=optenv32
+
+PKGEPOCH=1
```
Why bumping PKGEPOCH here?

MingcongBai replied:
Version changed from 1.14rc5 => 1.14~rc5, which is smaller.

OriginCode commented on runtime-optenv32/aosc-aaa+32/autobuild/preinst:
```diff
@@ -0,0 +1,5 @@
+for i in etc share var; do
+    if [ -L /opt/32/$i ]; then
+        rm /opt/32/$i
```
Add some verbosity.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/binutils+32/autobuild/defines:
```diff
@@ -1,12 +1,11 @@
 NOSTATIC=0
-NOLTO=1
 AB_FLAGS_O3=1
 RECONF=0
```
Missing ABHOST.

MingcongBai replied:
It's a few lines down at L34.

OriginCode commented on runtime-optenv32/binutils+32/autobuild/defines.stage2:
```diff
@@ -2,7 +2,6 @@ PKGNAME=binutils+32
 PKGSEC=devel
 PKGDEP="glibc zlib"
-NOLTO=1
 
```
Missing ABHOST.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/dbus+32/spec:
```diff
@@ -1,4 +1,5 @@
-SRCS="tbl::https://dbus.freedesktop.org/releases/dbus/dbus-$VER.tar.gz"
-CHKSUMS="sha256::64cf4d70840230e5e9bc784d153880775ab3db19d656ead8a0cb9c0ab5a95306"
+VER=1.14.10
+REL=3
```
Why bumping REL to 3?

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/elfutils+32/autobuild/defines:
```diff
@@ -1,6 +1,11 @@
+
+# Note: Minimise dependency.
+AUTOTOOLS_AFTER="--disable-debuginfod \
+                 --disable-libdebuginfod"
```
May use list format here as recommended, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/elfutils+32/spec:
```diff
@@ -1,5 +1,6 @@
-CHKSUMS="sha256::3c056914c8a438b210be0d790463b960fc79d234c3f05ce707cbff80e94cba30"
-REL=2
+VER=0.191
+REL=1
```
Why bumping REL here.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/enca+32/autobuild/defines:
```diff
@@ -1,10 +1,10 @@
+BUILDDEP="devel-base+32"
+PKGDES="Charset analyzer and converter (32-bit x86 runtime)"
 
 AUTOTOOLS_AFTER="--with-librecode=/usr --enable-external"
```
May use list format here, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/gcc+32/spec:
```diff
@@ -1,8 +1,9 @@
 __MPC_VER=1.3.1
 __MPFR_VER=4.2.1
 VER=$__GCC_VER
+REL=1
```
Why bumping REL.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/gdbm+32/autobuild/beyond:
```diff
@@ -1,6 +1,8 @@
-
-rm -rf $PKGDIR/opt/32/share/info/dir
+abinfo "Creating compatibility symlinks for gdbm headers ..."
+install -dm755 "$PKGDIR"/opt/32/include/gdbm
```
Add verbosity here.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/glibc+32/spec:
```diff
@@ -1,8 +1,8 @@
+VER=2.40
 LOCALEGENVER=20160914
 SRCS="tbl::https://ftp.gnu.org/gnu/glibc/glibc-$VER.tar.xz \
       https://github.com/AOSC-Dev/locale-gen/archive/refs/tags/v$LOCALEGENVER.tar.gz"
```
Use git?

MingcongBai replied:
Unused source, removed.

OriginCode commented on runtime-optenv32/gtk-2+32/autobuild/defines:
```diff
@@ -1,10 +1,12 @@
-ABSHADOW=no
-NOLTO=1
-ABHOST=noarch
+AUTOTOOLS_AFTER="--with-xinput=yes \
```
Use list, not mandatory.

MingcongBai replied:
Addressed (by the way, it's called an array).

OriginCode commented on runtime-optenv32/gtk-2+32/spec:
```diff
@@ -1,5 +1,5 @@
-VER=2.24.32
+VER=2.24.33
 REL=3
```
Remove the REL after updating.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/imlib2+32/spec:
```diff
@@ -1,5 +1,5 @@
-SRCS="tbl::https://sourceforge.net/projects/enlightenment/files/imlib2-src/$VER/imlib2-$VER.tar.bz2"
-CHKSUMS="sha256::89ab531ff882c23c8c68e3e941d1bb59dde7a3b2a7853b87ab8c7615da7cb077"
+VER=1.5.1
 REL=2
```
Remove the REL after updating.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/json-c+32/spec:
```diff
@@ -1,5 +1,4 @@
-VER=0.13.1+20180305
-REL=2
+VER=0.18
 SRCS="tbl::https://github.com/json-c/json-c/archive/json-c-${VER/+/-}.tar.gz"
```
Use git source.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/krb5+32/autobuild/defines:
```diff
@@ -1,9 +1,18 @@
+BUILDDEP="devel-base+32"
+PKGDES="The Kerberos network authentication system (32-bit x86 runtime)"
 
+AUTOTOOLS_AFTER="--enable-shared \
```
Use list, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/libass+32/autobuild/defines:
```diff
@@ -1,11 +1,10 @@
+PKGDEP="aosc-aaa+32 enca+32 fontconfig+32 fribidi+32"
+BUILDDEP="devel-base+32 yasm"
+
+AUTOTOOLS_AFTER="--enable-fontconfig"
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/libbluray+32/autobuild/defines:
```diff
@@ -1,9 +1,9 @@
-AUTOTOOLS_AFTER="--enable-udf --disable-examples --disable-bdjava-jar"
-NOLTO=1
-ABHOST=noarch
+AUTOTOOLS_AFTER="--disable-examples \
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/libcdio+32/autobuild/defines:
```diff
@@ -1,8 +1,9 @@
+BUILDDEP="devel-base+32"
+PKGDES="GNU compact disk input and control library (32-bit x86 runtime)"
 
 AUTOTOOLS_AFTER="--enable-cpp-progs"
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/libdv+32/autobuild/defines:
```diff
@@ -1,9 +1,11 @@
+BUILDDEP="autoconf-archive devel-base+32"
+PKGDES="The Quarsar DV codec (32-bit x86 runtime)"
 
 AUTOTOOLS_AFTER="--disable-gtk"
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/libjpeg-turbo+32/autobuild/build:
```diff
@@ -1,9 +1,40 @@
+done
+
+abinfo "Installing jpegint.h ..."
+install -vm644 "$SRCDIR"/jpegint.h \
+    "$PKGDIR"/opt/32/include
```
Do you assume the target directory exists? If not then add `-d` param here.

MingcongBai replied:
Oh definitely, this library definitely comes with headers.

OriginCode commented on runtime-optenv32/libssh+32/autobuild/defines:
```diff
@@ -1,9 +1,16 @@
+PKGDES="A client-side C library implementing the SSH protocol (32-bit x86 runtime)"
 
-CMAKE_AFTER="-DWITH_GSSAPI=ON -DWITH_ZLIB=/opt/32/lib"
+CMAKE_AFTER="-DWITH_GSSAPI=ON \
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/libva+32/autobuild/defines:
```diff
@@ -1,9 +1,10 @@
-NOLTO=1
-ABHOST=noarch
+ABTYPE=meson
+MESON_AFTER="-Dwith_wayland=no"
```
Use list format, not mandatory.

MingcongBai replied:
Addressed, also enabled Wayland support.

OriginCode commented on runtime-optenv32/libva-intel-driver+32/autobuild/defines:
```diff
@@ -1,8 +1,10 @@
-NOLTO=1
-ABHOST=noarch
+ABTYPE=meson
+MESON_AFTER="-Dwith_wayland=no"
```
Use list format, not mandatory.

MingcongBai replied:
Addressed, also enabled Wayland support.

OriginCode commented on runtime-optenv32/libxcrypt+32/autobuild/defines:
```diff
@@ -1,7 +1,7 @@
+PKGDES="Library for one-way hashing of passwords (32-bit x86 runtime)"
 BUILDDEP="devel-base+32"
 ABHOST=optenv32
 
```
Use list format, not mandatory. (below this line)

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/mpfr+32/autobuild/defines:
```diff
@@ -10,8 +10,8 @@ BUILDDEP="devel-base+32"
 # 2. Enables shared libraries (DO NOT disable, you will need this for shared GCC build);
 # 3. Declares documentation directory;
 AUTOTOOLS_AFTER="--enable-thread-safe \
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/nss+32/autobuild/certdata2pem.py:
What's this file for?

MingcongBai replied:
See prepare.

OriginCode commented on runtime-optenv32/p11-kit+32/autobuild/defines:
```diff
@@ -1,8 +1,10 @@
-ABHOST=noarch
+ABHOST=optenv32
+
+AUTOTOOLS_AFTER="--without-systemd \
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/pcre2+32/autobuild/defines:
```diff
@@ -0,0 +1,19 @@
+BUILDDEP="devel-base+32"
+PKGDES="Perl Compatible Regular Expression (2nd Edition, 32-bit x86 runtime)"
+
+AUTOTOOLS_AFTER="--enable-pcre2-16 \
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/popt+32/autobuild/defines:
```diff
@@ -1,11 +1,9 @@
+PKGDES="A command line option parser (32-bit x86 runtime)"
 
-NOSTATIC=no
 AUTOTOLS_AFTER="--enable-static"
```
AUTOTOLS 💯

MingcongBai replied:
Ayy, not necessary anyway LOL

OriginCode commented on runtime-optenv32/recode+32/autobuild/defines:
```diff
@@ -1,8 +1,9 @@
-AUTOTOOLS_AFTER="--without-included-gettext"
-NOLTO=1
-ABHOST=noarch
+AUTOTOOLS_AFTER="PYTHON=/usr/bin/python3"
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/rustc+32/spec:
```diff
@@ -0,0 +1,17 @@
+# to bootstrap from an adjacent release.
+#
+# Comment the following segment if this is not the case.
+#SRCS="tbl::https://static.rust-lang.org/dist/rustc-${VER}-src.tar.x \
```
`.tar.x` does not sound right to me.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/sdl-ttf+32/autobuild/defines:
```diff
@@ -1,7 +1,9 @@
+
+ABHOST=optenv32
+
+AUTOTOOLS_AFTER="--with-freetype-prefix=/opt/32"
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/shared-mime-info+32/autobuild/defines:
```diff
@@ -0,0 +1,10 @@
+PKGDES="Freedesktop.org shared MIME info (32-bit x86 runtime)"
+
+AUTOTOOLS_AFTER="--disable-update-mimedb"
+NOPARALLEL=1
```
Add some notes here.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/shared-mime-info+32/autobuild/defines:
```diff
@@ -0,0 +1,10 @@
+BUILDDEP="devel-base+32"
+PKGDES="Freedesktop.org shared MIME info (32-bit x86 runtime)"
+
+AUTOTOOLS_AFTER="--disable-update-mimedb"
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/soxr+32/autobuild/defines:
```diff
@@ -1,18 +1,16 @@
+BUILDDEP="devel-base+32"
+PKGDES="A library that aims to give fast and high quality results for any constant resampling ratio (32-bit x86 runtime)"
 
 CMAKE_AFTER="-DBUILD_EXAMPLES=OFF \
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/spirv-tools+32/autobuild/defines:
```diff
@@ -0,0 +1,15 @@
+BUILDDEP="python-3"
+PKGDES="API and commands for processing SPIR-V modules (32-bit x86 runtime)"
+
+CMAKE_AFTER="-DSPIRV-Headers_SOURCE_DIR=/opt/32 \
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/sqlite+32/autobuild/defines:
```diff
@@ -1,10 +1,9 @@
 
-NOLTO=1
 ABSHADOW=0
 NOPARALLEL=1
```
Add some notes here.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/sqlite+32/autobuild/defines:
```diff
@@ -1,10 +1,9 @@
-BUILDDEP="32subsystem"
+PKGDEP="aosc-aaa+32 readline+32 tcl+32"
+BUILDDEP="devel-base+32"
+PKGDES="A full-featured and self-contained SQL database engine (32-bit x86 runtime)"
```
Remove excessive descriptive words.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/tcl+32/autobuild/defines:
```diff
@@ -1,9 +1,8 @@
 PKGSEC=devel
+PKGDEP="aosc-aaa+32 zlib+32"
+BUILDDEP="devel-base+32"
+PKGDES="Tool Command Lanuage (32-bit x86 runtime)"
```
"Lanuage"

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/vulkan-headers+32/autobuild/defines:
```diff
@@ -1,10 +1,12 @@
-PKGDEP=""
-PKGDES="Vulkan header files and API registry (optenv32)"
-BUILDDEP="gcc+32"
+PKGDEP="aosc-aaa+32 "
```
Remove redundant space at the end.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/vulkan-tools+32/autobuild/defines:
```diff
@@ -1,8 +1,8 @@
+PKGDES="Vulkan Tools and Utilities (32-bit x86 runtime)"
 
 # FIXME: WSL_WAYLAND requires waylandpp+32
 CMAKE_AFTER="-DBUILD_CUBE=ON \
```
Use list format, not mandatory.

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/wavpack+32/autobuild/defines:
```diff
@@ -1,10 +1,9 @@
+BUILDDEP="devel-base+32"
+PKGDES="Audio compression format with lossless, lossy and hybrid compression modes (32-bit x86 runtime)"
 
 AUTOTOOLS_AFTER="--disable-asm"
```
Use list format, not mandatory.

MingcongBai replied:
This should not have been here in the first place.

OriginCode commented on runtime-optenv32/x11-lib+32/autobuild/defines.stage2:
```diff
@@ -0,0 +1,10 @@
+BUILDDEP="devel-base+32"
+PKGDES="X11 runtime libraries (32-bit x86 runtime)"
+
+PKGREP="libxp"
```
Does it really provide and replace a non-+32 package?

MingcongBai replied:
No, good catch.

OriginCode commented on runtime-optenv32/libvpx+32/autobuild/defines:
```diff
@@ -1,10 +1,9 @@
-PKGDEP="glibc+32"
-BUILDDEP="32subsystem"
+PKGDEP="aosc-aaa+32 glibc+32"
+BUILDDEP="devel-base+32 nasm"
```
Use `nasm+32` instead?

MingcongBai replied:
This proves exactly that nasm+32 was not needed. Dropped.

OriginCode commented on runtime-optenv32/sdl+32/autobuild/defines:
```diff
@@ -1,9 +1,12 @@
-PKGDEP="x11-lib+32 alsa-lib+32"
-BUILDDEP="32subsystem"
+PKGDEP="aosc-aaa+32 x11-lib+32 alsa-lib+32"
+BUILDDEP="devel-base+32 nasm"
```
Use `nasm+32` instead?

MingcongBai replied:
This proves exactly that nasm+32 was not needed. Dropped.

OriginCode commented on runtime-optenv32/x264+32/autobuild/defines:
```diff
@@ -1,10 +1,9 @@
-PKGDES="H264 encoding library (optenv32)"
+PKGDEP="aosc-aaa+32 glibc+32"
+BUILDDEP="devel-base+32 nasm+32"
+PKGDES="H264 encoding library (32-bit x86 runtime)"
```
H.264

MingcongBai replied:
Addressed.

Cyanoxygen commented:
<img width="695" alt="image" src="https://github.com/user-attachments/assets/1816d452-7085-4b8b-b6db-021d5d538fd1" />

It looks like there's no files installed to /opt/32/etc and /opt/32/share.

Cyanoxygen commented:
<img width="1388" alt="image" src="https://github.com/user-attachments/assets/f5c1f48e-86df-49d0-9343-77eb2ef9c2d2" />

Steam is able to start up in my VM. I am currently unable to test this PR with a physical machine, so please wait.

MingcongBai committed:
aosc-aaa+32: prompt about removal of shared symlinks

MingcongBai committed:
binutils+32: (stage2) mark as ABHOST=optenv32

MingcongBai committed:
dbus+32: drop extraneous REL=

MingcongBai committed:
elfutils+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
elfutils+32: drop extraneous REL=

MingcongBai committed:
enca+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
gcc+32: drop extraneous REL=

MingcongBai committed:
gdbm+32: add missing verbosity

MingcongBai committed:
glibc+32: drop unused locale-gen source

MingcongBai committed:
gtk-2+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
gtk-2+32: drop extraneous REL=

MingcongBai committed:
imlib2+32: drop extraneous REL=

MingcongBai committed:
json-c+32: use Git source and UPSTREAM_VER

MingcongBai committed:
krb5+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
libass+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
libbluray+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
libcdio+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
libdv+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
libssh+32: use array for CMAKE_AFTER

MingcongBai committed:
libva-intel-driver+32: enable Wayland support

MingcongBai committed:
libva+32: enable Wayland support

MingcongBai committed:
libxcrypt+32: use array for AUTOTOOLS_AFTER*

MingcongBai committed:
mpfr+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
p11-kit+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
pcre2+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
popt+32: drop unneeded AUTOTOOLS_AFTER

MingcongBai committed:
recode+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
rustc: fix typo .tar.x => .tar.xz

MingcongBai committed:
shared-mime-info+32: add comments about hacks and fixes

MingcongBai committed:
sdl-ttf+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
shared-mime-info+32: use array for AUTOTOOLS_AFTER

MingcongBai committed:
soxr+32: use array for CMAKE_AFTER

MingcongBai committed:
spirv-tools+32: use array for CMAKE_AFTER

MingcongBai committed:
sqlite+32: add comments for fixes and hacks

MingcongBai committed:
sqlite+32: drop advertisement-like description

MingcongBai committed:
tcl+32: fix a typo Lanuage => Language

MingcongBai committed:
vulkan-headers+32: drop trailing space in PKGDEP

MingcongBai committed:
vulkan-tools+32: use array for CMAKE_AFTER

MingcongBai committed:
wavpack+32: drop unneeded AUTOTOOLS_AFTER

MingcongBai committed:
x11-lib+32: drop invalid PKGREP/PKGPROV

MingcongBai committed:
nasm+32: drop, orphaned

MingcongBai committed:
x264+32: fix a typo H264 => H.264

OriginCode commented on runtime-optenv32/elfutils+32/spec:
```diff
@@ -1,5 +1,5 @@
-REL=2
+VER=0.191
+SRCS="tbl::https://sourceware.org/elfutils/ftp/$VER/elfutils-$VER.tar.bz2"
+# checksum: https://sourceware.org/elfutils/ftp/$VER/sha512.sum
+CHKSUMS="sha256::df76db71366d1d708365fc7a6c60ca48398f14367eb2b8954efc8897147ad871"
```
Is it sha512 or sha256?

MingcongBai replied:
Addressed.

OriginCode commented on runtime-optenv32/glibc+32/spec:
```diff
@@ -1,8 +1,5 @@
-VER=2.38
+VER=2.40
 LOCALEGENVER=20160914
```
Remove the extraneous variable.

MingcongBai replied:
Addressed.

Cyanoxygen committed:
json-c: use the correct tag name with release date

The name of the tagged release follows the format of $VER-$DATE.

Cyanoxygen committed:
rustc: update to 1.85.0

MingcongBai committed:
v4l-utils+32: disable shadow build

FIXME: make[2]: *** No rule to make target /config.status, needed by Makefile.  Stop.

MingcongBai committed:
elfutils+32: drop inaccurate comment about checksums

MingcongBai committed:
glibc+32: drop unused LOCALEGENVER

MingcongBai committed:
v4l-utils+32: disable RECONF instead of the ineffective ABSHADOW=0

FIXME: make[2]: *** No rule to make target '/config.status', needed by 'Makefile'.  Stop.

MingcongBai committed:
v4l-utils+32: add mesa+32 dep

MingcongBai committed:
v4l-utils+32: add glu+32 dep

MingcongBai committed:
x264+32: use nasm from main system

OriginCode commented:
Smoke test with Steam passed
![image](https://github.com/user-attachments/assets/48688b8e-cce9-48ff-86ee-5a4427a823cf)

OriginCode commented on runtime-optenv32/x264+32/autobuild/defines:
```diff
@@ -1,10 +1,9 @@
-AUTOTOOLS_AFTER="--enable-shared --enable-pic"
-NOLTO=1
-ABHOST=noarch
+PKGEPOCH=1
```
Why bumping EPOCH here?

MingcongBai replied:
```
x264+32/optenv32-revamp-20241116,now 1:0+git20241027~pre20250302T060954Z amd64 [installed,automatic]
x264+32/stable 20160827.2245-3 all [automatic]
```

MingcongBai commented:
Tested good with L4D2:

https://github.com/user-attachments/assets/31d34a44-fb43-41f3-bd0f-a89486edf350
